docs/docs/life-and-times-of-a-gatsby-build.md

```diff
@@ -2,7 +2,305 @@
+
+Runs the [`onPreInit` node API](/docs/node-apis/#onPreInit) if it has been implemented by your site or any installed plugins.
+
+4. `delete html and css files from previous builds`
+
+The only different step between develop and build, the HTML and CSS from previous builds is deleted to prevent problems with styles and pages that no longer exist sticking around.
+
+5. `initialize cache`
+
+Check if new dependencies have been installed in the `package.json`, if the versions of installed plugins have changed, or if the `gatsby-config.js` or the `gatsby-node.js` files have changed. Plugins can [interact with the cache](/docs/build-caching/).
+
+6. `copy gatsby files`
+
+Copies site files into the cache in the `.cache` folder.
+
+7. `onPreBootstrap`
+
+Calls the [`onPreBootstrap` node API](/docs/node-apis/#onPreBootstrap) in your site or plugins where it is implemented.
+
+8. `source and transform nodes`
+
+Creates nodes from your site and all plugins implementing the [`sourceNodes` API](/docs/node-apis/#sourceNodes), and warns about plugins that aren't creating any nodes. Nodes created by source or transformer plugins are cached.
+
+Nodes created at this stage are considered top level nodes, meaning they don't have a parent node that they are derived from.
+
+9. `Add explicit types`
+
+Adds types to the GraphQL schema for nodes that you have defined explicitly with Gatsby's [schema optimization APIs](/docs/schema-customization/#explicitly-defining-data-types).
+
+10. `Add inferred types`
+
+All other nodes not already defined are inspected and have types [inferred](/docs/schema-customization/#automatic-type-inference) by Gatsby.
+
+11. `Processing types`
+
+Composes 3rd party schema types, children fields, custom resolve functions, set fields, and print type definitions into the GraphQL schema.
+
+12. `building schema`
+
+Imports the composed GraphQL schema and builds it.
+
+13. `createPages`
+
+Calls the [`createPages` API](/docs/node-apis/#createPages) for your site and all plugins implementing it, like when you [create pages programatically](/docs/programmatically-create-pages-from-data/) in your `gatsby-node.js`.
+
+Plugins can handle the [`onCreatePage` event](/docs/node-apis/#onCreatePage) at this point for use cases like manipulating the path of pages.
+
+14. `createPagesStatefully`
+
+Similar to the `createPages` step, but for the [`createPagesStatefully` API](/docs/node-apis/#createPagesStatefully).
+
+15. `onPreExtractQueries`
+
+Calls the [`onPreExtractQueries` API](/docs/node-apis/#onPreExtractQueries) for your site and all plugins implementing it.
+
+16. `update schema`
+
+Rebuilds the GraphQL schema, this time with `SitePage` context.
+
+17. `extract queries from components`
+
+All JavaScript files in the site are loaded and Gatsby determines if there are any GraphQL queries in them. If there are problematic queries they can be reported back with warnings or errors. All these queries get queued up for execution in a later step.
```

Comment:
```suggestion
All JavaScript files in the site are loaded and Gatsby determines if there are any GraphQL queries in them as exports. If there are problematic queries they can be reported back with warnings or errors. All these queries get queued up for execution in a later step.
```

i remember the name of the export is not interesting, and only one query per JavaScript file?